plugin/PlayAssetDelivery/src/main/java/com/google/play/core/godot/assetpacks/PlayAssetDelivery.java

```diff
@@ -63,12 +62,45 @@ public PlayAssetDelivery(Godot godot) {
     super(godot);
     Context applicationContext = godot.getApplicationContext();
     assetPackManager = AssetPackManagerFactory.getInstance(applicationContext);
+    registerAssetPackStateUpdatedListener();
   }
 
   /** Package-private constructor used to instantiate PlayAssetDelivery class with mock objects. */
   PlayAssetDelivery(Godot godot, AssetPackManager assetPackManager) {
     super(godot);
     this.assetPackManager = assetPackManager;
+    registerAssetPackStateUpdatedListener();
+  }
+
+  @Override
+  public void onMainPause() {
+    assetPackManager.clearListeners();
+    super.onMainPause();
+  }
+
+  @Override
+  public void onMainResume() {
+    registerAssetPackStateUpdatedListener();
```

Comment:
Is there a possibility that this class will register this listener twice? Once in the constructor and again during onMainResume?

Reply:
I double checked the Activity Lifecycle and indeed ```registerAssetPackStateUpdatedListener()``` will be called twice. Fixed this bug and removed the registerAssetPackStateUpdatedListener() in constructor.